<launch>
    <!-- Simulator Configuration -->
    <arg name="simulation" value="False"/>

    <!-- Trajectory Configuration -->
    <arg name="start_frame" value="ar_marker_2"/>
    <arg name="end_frame" value="ar_marker_14"/>

    <!-- Robot 1 Configuration -->
    <arg name="robot1_frame" value="ar_marker_5"/>
    <arg name="robot1_config" value="0"/>
    <arg name="robot1_control" value="/pink/cmd_vel_mux/input/teleop"/>

    <!-- Robot 2 Configuration -->
    <arg name="robot2_frame" value="ar_marker_4"/>
    <arg name="robot2_config" value="0.5"/>
    <arg name="robot2_control" value="/green/cmd_vel_mux/input/teleop"/>

    <!-- Frame Publisher Node -->
    <node name="frame_publisher" pkg="balebot" type="frame_publisher.py"  output="screen"/>

    <!-- Sensing Node -->
    <node name="state_observer" pkg="balebot" type="state_observer.py" output="screen">
        <param name="simulation" value="$(arg simulation)" type="bool"/>
        <param name="start_frame" value="$(arg start_frame)" type="string"/>
        <param name="end_frame" value="$(arg end_frame)" type="string"/>
        <param name="robot1_frame" value="$(arg robot1_frame)" type="string"/>
        <param name="robot2_frame" value="$(arg robot2_frame)" type="string"/>
        <param name="robot1_config" value="$(arg robot1_config)" type="double"/>
        <param name="robot2_config" value="$(arg robot2_config)" type="double"/>
    </node>

    <!-- Following Node -->
    <node name="follow_controller" pkg="balebot" type="follow_controller.py" output="screen">
        <param name="robot1_control" value="$(arg robot1_control)" type="string"/>
        <param name="robot2_control" value="$(arg robot2_control)" type="string"/>
        <param name="robot1_config" value="$(arg robot1_config)" type="double"/>
        <param name="robot2_config" value="$(arg robot2_config)" type="double"/>
    </node>

</launch>
